### PR TITLE
Update typescript definitions

### PR DESCRIPTION
**Pull request by hoetmaaiers**
I have added my local typescript definitions to match the documation and my own usage. This is far from complete but just an update for my current project.

Referring: https://github.com/geoman-io/leaflet-geoman/issues/790

The event system is harder to type and is added in comment. Any help is surely welcome.

### CODE REVIEW

**Comment by Falke-Design** [comment]
@hoetmaaiers can you please resolve the merge conflict and pull the newest develop version? We will create a release this week and it would be nice to have this PR in

**Review by Falke-Design — COMMENTED** [review]
Maybe you want to update this changes but it is not necessary to merge this PR

File: leaflet-geoman.d.ts [hunk 1]
```diff
@@ -1,164 +1,185 @@
...
+        cutButton?: string;
+        deleteButton?: string;
+        drawCircleMarkerButton?: string;
+      };
     }
 
-    interface LayerOptions {
-        pmIgnore?: boolean
+    interface Map {
+      addControls(options?: DrawControlOptions): void;
```
Comment (Falke-Design) [inline leaflet-geoman.d.ts:135]: - controlsVisible()
- globalCutModeEnabled
- enableGlobalCutMode
- toggleGlobalCutMode
- disableGlobalCutMode
- globalEditModeEnabled
- enableGlobalDragMode
- disableGlobalDragMode
- globalDragModeEnabled
- toggleGlobalDragMode
- enableGlobalRemovalMode
- disableGlobalRemovalMode
- globalRemovalModeEnabled
- toggleGlobalRemovalMode

#821

File: leaflet-geoman.d.ts [hunk 2]
```diff
@@ -1,164 +1,185 @@
...
+    pm: PM.Layer;
+  }
+
+  interface LayerOptions {
+    pmIgnore?: boolean;
+  }
+
+  interface MapOptions {
+    pmIgnore?: boolean;
+  }
```
Comment (Falke-Design) [inline leaflet-geoman.d.ts:184]: Interface PM
- reInitLayer()
- setOptIn

**Comment by codeofsumit** [comment]
Closing this one in favor of #825

**Pull request closed without merging**